src/code_8097670.c

```diff
@@ -201,4 +210,203 @@ bool8 sub_8097900(s16 pokeIndex)
     if (preload->unk1C[iVar2 >> 5] & (1 << (baseSpecies + (iVar2 >> 5) * -0x20)))
         return TRUE;
     return FALSE;
-}
\ No newline at end of file
+}
+
+void sub_8097944(void)
+{
+    bool8 bVar2;
+    bool8 bVar3;
+    int index;
+    s32 counter;
+    Move *move;
+    PokemonStruct1 *pokeStruct;
+    s32 index1;
+    s32 temp;
+
+    bVar2 = 1;
+    bVar3 = 1;
+    if ((sub_8097710(0xd) == 0) && (HasAllFriendAreas())) {
+        sub_80976F8(0xd);
+    }
+
+    for(index = 0; index < NUM_MONSTERS; index++)
+    {
+
+#ifdef NONMATCHING
+        pokeStruct = &index[gRecruitedPokemonRef->pokemon];
+#else
+        register size_t offset asm("r1") = offsetof(unkStruct_203B45C, pokemon[index]);
+        PokemonStruct1* p = gRecruitedPokemonRef->pokemon;
+        size_t addr = offset + (size_t)p;
+        pokeStruct = (PokemonStruct1*)addr;
+#endif
+
+
+        if (((u8)(pokeStruct->unk0) & 1)) {
+            temp = pokeStruct->speciesNum;
+            gUnknown_203B494->unk1C[(temp / 32)] |= (1 << (temp + (temp / 32) * -32));
+
+            for(index1 = 0; index1 < MAX_MON_MOVES; index1++)
+            {
+                move = &gRecruitedPokemonRef->pokemon[index].moves[index1];
+                if ((move->moveFlags & 1) != 0) {
+                    gUnknown_203B494->unk8C[move->id / 32] |= 1 << (move->id & 0x1f);
+                }
+            }
+        }
+  }
+    
+  for(index = 0; index < MONSTER_MAX; index++)
+  {
+    if (index != MONSTER_NONE)
```

Comment:
This look terrible lol. Are you sure this couldn't be changed to something else? `continue` maybe?

Reply:
Haven't looked at it in a while but here's the scratch link:

https://decomp.me/scratch/wH4qc

Reply:
At the very least the horrible if() if() if() dance can be converted to a single if (.. && .. && .. && ..)

Reply:
Here's a cleaned up scratch, had a go at the NONMATCHING but couldn't get it to go away https://decomp.me/scratch/WyTn5